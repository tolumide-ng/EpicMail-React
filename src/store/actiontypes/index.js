export const SIGNUP_PENDING = 'SIGNUP_PENDING';

export const SIGNUP_FAILURE = 'SIGNUP_FAILURE';

export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';

export const LOGIN_PENDING = 'LOGIN_PENDING';

export const LOGIN_FAILURE = 'LOGIN_FAILURE';

export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';

export const REQUEST_RESET_PENDING = 'REQUEST_RESET_PENDING';

export const REQUEST_RESET_FAILURE = 'REQUEST_RESET_FAILURE';

export const REQUEST_RESET_SUCCESS = 'REQUEST_RESET_SUCCESS';
