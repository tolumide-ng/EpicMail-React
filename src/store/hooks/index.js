import { useEffect } from 'react';
import { getUser } from '../actions/Signup';

export const useSetUser = ({ dispatch }) => {
  useEffect(() => {
    dispatch(getUser());
  }, []);
  return [];
};

export default useSetUser;
