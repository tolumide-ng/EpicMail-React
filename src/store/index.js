import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import rootReducer from './reducers';


const middlewares = applyMiddleware(thunk);

const store = createStore(
  rootReducer,
  composeWithDevTools(middlewares)
  );


export default store;