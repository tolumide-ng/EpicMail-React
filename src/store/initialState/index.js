import { auth } from './auth';
import { composeMail } from './composeMail';
import { composeDraft } from './composeDraft';
import { sentMessages } from './sentMessages';
import { requestReset } from './requestReset';
import { deleteMessage } from './deleteMessage';
import { specificSentMessage } from './specificSentMessage';
import { retractMessage } from './retractMessage';

const initialState = {
  auth,
  requestReset,
  composeMail,
  composeDraft,
  sentMessages,
  deleteMessage,
  specificSentMessage,
  retractMessage
};

export default initialState;
