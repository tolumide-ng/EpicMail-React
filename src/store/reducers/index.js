import { combineReducers } from 'redux';
import authReducer from './auth';
import requestResetReducer from './requestReset';
import composeMailReducer from './composeMail';
import composeDraftReducer from './composeDraft';
import sentMessagesReducer from './sentMessages';
import deleteMessageReducer from './deleteMessage';
import specificSentMessageReducer from './specificSentMessage';
import retractMessageReducer from './retractMessage';

const rootReducer = combineReducers({
  authReducer,
  requestResetReducer,
  composeMailReducer,
  composeDraftReducer,
  sentMessagesReducer,
  deleteMessageReducer,
  specificSentMessageReducer,
  retractMessageReducer
});

export default rootReducer;
