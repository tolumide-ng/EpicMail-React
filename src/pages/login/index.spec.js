import React from 'react';
import { render, fireEvent, cleanup, wait } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import configureMockStore from 'redux-mock-store';

import { LoginForm, mapStateToProps, mapDispatchToProps } from './index';
import store from '../../store';

// import jest from 'jest';

describe('LoginForm Validation Fails', () => {
  afterEach(cleanup);
  test('should have validation error when username field is touched and not complete', async () => {
    const placeholderName = 'username';
    const { asFragment, getByPlaceholderText, findByTestId } = render(
      <LoginForm store={store} />
    );
    const input = getByPlaceholderText(placeholderName);
    expect(asFragment).toMatchSnapshot();
    fireEvent.blur(input);
    const validationErrors = await findByTestId('usernameError');
    expect(validationErrors.innerHTML).toBe('username is required');
  });

  test('should have validation error when password field is touched and not complete', async () => {
    const { asFragment, getByPlaceholderText, findByTestId } = render(
      <LoginForm store={store} />
    );
    const input = getByPlaceholderText('password');
    expect(asFragment).toMatchSnapshot();
    fireEvent.blur(input);
    const validationErrors = await findByTestId('passwordError');
    expect(validationErrors.innerHTML).toBe('password is required');
  });
});

describe('LoginForm Validation Success', () => {
  afterEach(cleanup);
  test('simulate input and click the form submit button', async () => {
    const login = jest.fn();

    const { asFragment, getByPlaceholderText, getByTestId, getByText } = render(
      <LoginForm store={store} login={login} lo="tese" />
    );
    const usernameInput = getByPlaceholderText('username');
    const passwordInput = getByPlaceholderText('password');
    const submitButton = getByTestId('submitButton');
    const username = 'tolumideusername@epicmail.com';
    const password = 'Password2019#';

    fireEvent.change(usernameInput, { target: { value: username } });
    expect(usernameInput.value).toBe(username);

    fireEvent.change(passwordInput, { target: { value: password } });
    expect(passwordInput.value).toBe(password);

    expect(getByText('Sign in now')).toBeTruthy();
    fireEvent.submit(submitButton);
    expect(getByText('Signing in')).toBeTruthy();
    expect(asFragment).toMatchSnapshot();
    await wait(() => {
      expect(login).toHaveBeenCalled();
    });
  });
});

describe('Dispatch actions on validation success', () => {
  afterEach(cleanup);

  test('should show initial state', () => {
    const initialState = {
      error: 'Authentication error'
    };
    expect(mapStateToProps(initialState).error).toEqual('Authentication error');
  });

  test('should dispatch login request action', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).login({});
  });
});
