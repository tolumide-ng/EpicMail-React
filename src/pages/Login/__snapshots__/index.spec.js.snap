// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Validation Fails should have validation error when password field is touched and not complete 1`] = `[Function]`;

exports[`LoginForm Validation Fails should have validation error when username field is touched and not complete 1`] = `[Function]`;

exports[`LoginForm Validation Success simulate input and click the form submit button 1`] = `[Function]`;
